adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/CoverageRDD.scala

```diff
@@ -59,17 +62,52 @@ private[adam] class CoverageArraySerializer(kryo: Kryo) extends IntervalArraySer
   }
 }
 
-object CoverageRDD {
+case class ParquetUnboundCoverageRDD private[rdd] (
+    private val sc: SparkContext,
+    private val parquetFilename: String,
+    sequences: SequenceDictionary) extends CoverageRDD {
 
-  /**
-   * Builds a CoverageRDD that does not have a partition map.
-   *
-   * @param rdd The underlying Coverage RDD.
-   * @param sd The SequenceDictionary for the RDD.
-   * @return A new Coverage RDD.
-   */
-  def apply(rdd: RDD[Coverage], sd: SequenceDictionary): CoverageRDD = {
-    CoverageRDD(rdd, sd, None)
+  lazy val rdd: RDD[Coverage] = {
+    sc.loadParquetCoverage(parquetFilename,
+      forceRdd = true).rdd
+  }
+
+  protected lazy val optPartitionMap = sc.extractPartitionMap(parquetFilename)
+
+  lazy val dataset = {
+    val sqlContext = SQLContext.getOrCreate(sc)
+    import sqlContext.implicits._
+    sqlContext.read.parquet(parquetFilename)
+      .select("contigName", "start", "end", "score")
+      .withColumnRenamed("score", "count")
+      .as[Coverage]
+  }
+
+  def toFeatureRDD(): FeatureRDD = {
+    ParquetUnboundFeatureRDD(sc, parquetFilename, sequences)
```

Comment:
I mean, I suppose you could do `coverage.rdd.cache()`. We should probably add cache/persist methods onto `GenomicRDD` that clarify this behavior (similar to what GraphX has).